src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/MethodTypeInference.cs

```diff
@@ -2443,13 +2440,13 @@ private bool Fix(int iParam, ref bool hadNullabilityMismatch, ref HashSet<Diagno
             return best;
         }
 
-        private NullableAnnotation GetNullableAnnotation(BoundExpression expr)
+        private TypeSymbolWithAnnotations GetTypeWithAnnotations(BoundExpression expr)
         {
-            if (!_conversions.IncludeNullability)
+            if (_conversions.IncludeNullability && _getTypeWithAnnotationOpt != null)
             {
-                return NullableAnnotation.Unknown;
+                return _getTypeWithAnnotationOpt.Invoke(expr);
```

Comment:
>_getTypeWithAnnotationOpt.Invoke(expr); [](start = 23, length = 39)

Minor thing: Since conditional access is no longer needed, we can drop the ```.Invoke```